include/onnxruntime/core/framework/allocator.h

```diff
@@ -44,6 +61,11 @@ class IAllocator {
   virtual void Free(void* p) = 0;
   const OrtMemoryInfo& Info() const { return memory_info_; };
 
+  /**
+     Some allocators may optionally choose to do some stateful cleanup after every Run() call
+  */
+  virtual Status OnRunEnd() { return Status::OK(); }
```

Comment:
Rename this method to something else. OnRunEnd() is the context in which this is getting called; doesn't indicate what it is doing.

Reply:
As indicated above, I ll rename this to just Shrink().

Reply:
Took down OnRunEnd() in IAllocator and exposed Shrink() in IArenaAllocator. Shrink() is relevant only to arena allocators anyway (similar to Reserve() in IArenaAllocator). I think it will be cleaner this way.